apps/st2-history/controller.js

```diff
@@ -253,6 +256,46 @@ angular.module('main')
       }
     };
 
+    $scope.rerun = (function () {
+      var rerun = $scope.$new();
+
+      rerun.metaSpec = {
+        type: 'object',
+        properties: {
+          ref: {
+            type: 'string',
+            name: 'Action'
+          }
+        }
+      };
+
+
+      rerun.open = function () {
+        $scope.$root.state.go('^.rerun', {id: $scope.record.id});
+        rerun.payload = _.clone($scope.payload);
```

Comment:
Actually, sorry, I'm slightly confused about this example:

> So, say we have an execution of core.local with parameters {cmd: 'ls -lah', cwd: '/'}. The rerun.payload would be the same as original payload. Then user would change cwd to /home and the resulting payload sent to the server would be {cmd: 'ls -lah', cwd: '/'}.

I assume here you meant "Then user would change cwd to /home and the resulting payload sent to the server would be {cmd: 'ls -lah', cwd: **'/home'**}."

As far as using default value goes - you would need to retrieve a default value from runner types API. Unless the default value is empty (None), in this case you are correct, you can't do that right now.

I can't think of a good approach to handle that - I'm against using a special string and I also don't like the approach of including additional attribute in the payload.